test/e2e/kubectl/kubectl.go

```diff
@@ -2069,6 +2072,82 @@ metadata:
 			gomega.Expect(outBuiltIn).To(gomega.Equal(outStatusSubresource))
 		})
 	})
+
+	ginkgo.Describe("Kubectl rollout", func() {
+		deploymentYaml := commonutils.SubstituteImageName(string(readTestFileOrDie(labeledDeploymentFilename)))
+		ginkgo.AfterEach(func() {
+			cleanupKubectlInputs(deploymentYaml, ns, "labels_foo=labels_bar")
+		})
+		ginkgo.It("undo should rollback and update deployment env", func(ctx context.Context) {
+			// create deployment
+			_ = e2ekubectl.RunKubectlOrDieInput(ns, deploymentYaml, "apply", "-f", "-")
+
+			d, err := c.AppsV1().Deployments(ns).Get(ctx, "pause-deployment", metav1.GetOptions{})
+			if err != nil {
+				framework.Failf("get deployment failed")
+			}
+			if err := e2edeployment.WaitForDeploymentComplete(c, d); err != nil {
+				framework.Failf("created deployment not ready")
+			}
+
+			origEnv := d.Spec.Template.Spec.Containers[0].Env
```

Comment:
you should verify here that `foo` and `var` are not set just to be defensive, we rely on a manifest that is in a folder and can be modified later for other tests